File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.js
function solution(n, k) {
    
    const str = n.toString(k);
    
    const arr = str.split("0");
    
    var answer = 0;
    
    for(let i=0; i<arr.length; i++){
        if(arr[i].length && arr[i] != "1"){
            if(isPrime(arr[i])) answer++;
        }
    }
    
    return answer;
}

function isPrime(str){
    
    const num = str;
    
    for(let i=2; i<= Math.sqrt(num); i++){
        if(num % i == 0) return false;
    }
    
    return true;
    
}

File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.java
import java.util.*;

class Solution {
    public int solution(int n, int k) {
        
        String change = Integer.toString(n, k);
        
        String arr[] = change.split("0");
        
        int answer = 0;
        
        for(int i=0; i<arr.length; i++){
            if(!arr[i].isEmpty() && !arr[i].equals("1")){
                if(isPrime(arr[i])) answer++;
            }
        }
        
        return answer;
    }
    
    public boolean isPrime(String str){
        
        long num = Long.parseLong(str);
        
        for(int i=2; i<=Math.sqrt(num); i++){
            if(num%i == 0) return false;
        }
        
        return true;
    }
}